src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -305,23 +305,32 @@ void Terminal::UpdateSelection(SelectionDirection direction, SelectionExpansion
         break;
     }
 
+    // 2.B) Clamp the movement to the mutable viewport
+    const auto bufferSize = _activeBuffer().GetSize();
+    const auto mutableViewport = _GetMutableViewport();
+    const COORD bottomRightInclusive{ mutableViewport.RightInclusive(), mutableViewport.BottomInclusive() };
+    if (bufferSize.CompareInBounds(targetPos, bottomRightInclusive) > 0)
+    {
+        targetPos.Y = bottomRightInclusive;
+    }
+
     // 3. Actually modify the selection
     // NOTE: targetStart doesn't matter here
     auto targetStart = false;
     std::tie(_selection->start, _selection->end) = _PivotSelection(targetPos, targetStart);
 
     // 4. Scroll (if necessary)
-    if (const auto viewport = _GetVisibleViewport(); !viewport.IsInBounds(targetPos))
+    if (const auto visibleViewport = _GetVisibleViewport(); !visibleViewport.IsInBounds(targetPos))
     {
-        if (const auto amtAboveView = viewport.Top() - targetPos.Y; amtAboveView > 0)
+        if (const auto amtAboveView = visibleViewport.Top() - targetPos.Y; amtAboveView > 0)
         {
             // anchor is above visible viewport, scroll by that amount
             _scrollOffset += amtAboveView;
         }
         else
         {
             // anchor is below visible viewport, scroll by that amount
-            const auto amtBelowView = targetPos.Y - viewport.BottomInclusive();
+            const auto amtBelowView = targetPos.Y - visibleViewport.BottomInclusive();

```

Comment:
These changes appear unnecessary right?

Reply:
Eh, they're more helpful on `main` than here. I just wanted to create more clarity to distinguish between the visible viewport and mutable viewport. I'll still keep it at least to keep things as consistent as possible between these two branches.